### DISCUSSION TITLE
Docker install on Synology - teslamate cannot connect to database(postgres)

### DISCUSSION THREAD

**Discussion by safetyguy14** [teslamate-org/teslamate#1764] [Getting started]
**Describe the bug**

I attempted to install the latest of TeslaMate using the latest settings for docker-compose.yml (https://docs.teslamate.org/docs/installation/docker/).  All 4 packages install and start running without issue.  The database log shows the database is ready to accept connections.

**Expected behavior**

Teslamate should connect to the database and begin running the backend for the web interface.

**How to reproduce it (as minimally and precisely as possible)**:

1. SSH into Synology NAS, obtain root, and navigate to @docker
2. install teslamate using docker-compose up -d

![docker_database_1](https://user-images.githubusercontent.com/25259031/123552220-cb129600-d73a-11eb-87a6-b511c0bc74aa.jpg)

![docker_teslamate_1](https://user-images.githubusercontent.com/25259031/123641466-e8ea0480-d7e7-11eb-997c-2af962ca8686.jpg)

![docker-compose ps](https://user-images.githubusercontent.com/25259031/123641486-ec7d8b80-d7e7-11eb-9970-d2d6a0075c08.JPG)

**Environment**
- TeslaMate version: teslamate/teslamate:latest, postgres:13, teslamate/grafana:latest, eclipse-mosquitto:2
- Type of installation: Docker
- OS TeslaMate is installed on: Synology NAS - Docker 18.09.0-0519
- User OS & Browser: Windows 10 & Microsoft Edge

**Comment by Orbedk** [comment]
@safetyguy14 I just wanna make sure that you have seen this: https://github.com/adriankumpf/teslamate/issues/113
I followed that guide and it worked perfectly for me.

**Reply by safetyguy14** [reply]
I followed those instructions save for the changing the VIRTUAL_HOST and TZ environment variables since they are no longer in the latest docker-compose.

**Comment by adriankumpf** [comment]
Check out the TeslaMate container logs to see why it cannot connect to the database. I’ll move this to a discussion as there is no indication of a TeslaMate bug.

**Reply by safetyguy14** [reply]
I thought I had uploaded it in the bug report, sorry:

![docker_teslamate_1](https://user-images.githubusercontent.com/25259031/123633163-7a547900-d7de-11eb-8d63-4d87562eed99.jpg)

**Reply by DrMichael** [reply]
What says the database log? What `docker-compose ps`?

**Reply by safetyguy14** [reply]
database log is in the original comment; attached below the docker-compose ps
![docker-compose ps](https://user-images.githubusercontent.com/25259031/123641317-c1933780-d7e7-11eb-8128-090e87ae0cb6.JPG)

**Reply by DrMichael** [reply]
Looks all good. Just curious, what gives `netstat -tulpen | grep 5432` on the host?

**Reply by safetyguy14** [reply]
![netstat](https://user-images.githubusercontent.com/25259031/123657273-44bb8a00-d7f6-11eb-81f0-6c26c61bff7b.JPG)

**Reply by DrMichael** [reply]
Ok, there is a postgres database already running on the system, but should not doing any harm as long as you have
```
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS=secret
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
```
in the teslamate environment section.

**Reply by safetyguy14** [reply]
![teslamate_env_var](https://user-images.githubusercontent.com/25259031/123659939-cf04ed80-d7f8-11eb-87b0-afc495131df1.JPG)

**Reply by DrMichael** [reply]
Nice, can you go into the teslamate container and `ping database`?

**Reply by safetyguy14** [reply]
that's apparently a bridge to far for my docker knowledge, searching around for how to do that, have tried docker-compose run docker_teslamate_1 ping database but doesn't work

**Reply by DrMichael** [reply]
Ooops, the teslamate container does not have a ping...:-(

**Comment by BengalPrint** [comment]
Can anyone provide a dummies guide to backing up/restoring on Synology/Docker for Telsmate, and more importantly. How do I update my Teslamate on Docker (on Synology). Thanks!

**Comment by windy1lee** [comment]
Have you fixed it? I have a Synology 920+, and I encountered the same issue during installation – unable to connect to the database!

**Reply by safetyguy14** [reply]
Never did figure it out and gave up
